src/SnD.Sdk.Storage.Minio/MinioApiExtension.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Polly;
+
+namespace SnD.Sdk.Storage.Minio;
+
+/// <summary>
+/// Extension methods for Minio API.
+/// </summary>
+public static class MinioApiExtension
+{
+    /// <summary>
+    /// Applies a timeout retry policy to the specified asynchronous method.
+    /// </summary>
+    /// <typeparam name="TResult">The type of the result produced by the asynchronous method.</typeparam>
+    /// <typeparam name="TCaller">The type of the caller, used for logging purposes.</typeparam>
+    /// <param name="asyncMethod">The asynchronous method to which the retry policy will be applied.</param>
+    /// <param name="retryLogger">The logger used to log retry attempts.</param>
+    /// <param name="cancellationToken">An optional cancellation token to cancel the operation.</param>
+    /// <returns>A task that represents the asynchronous operation, which produces the result of the wrapped asynchronous method.</returns>
+    public static async Task<TResult> WithTimeoutRetryPolicy<TResult, TCaller>(this Func<CancellationToken, Task<TResult>> asyncMethod,
```

Comment:
I would suggest to make an exponential backoff with hardcoded start (like, 1 second or 0.5 seconds): https://github.com/App-vNext/Polly/wiki/Retry#exponential-backoff